packages/flutter_test/lib/src/binding.dart

```diff
@@ -700,13 +751,43 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
     // The LiveTestWidgetsFlutterBinding overrides this to report the exception to the console.
   }
 
+  Future<dynamic> _handleMessage(dynamic mockMessage) async {
+    final Map<dynamic, dynamic> message = mockMessage as Map<dynamic, dynamic>;
+    if (message['type'] == 'announce') {
+      final Map<dynamic, dynamic> data =
+          message['data'] as Map<dynamic, dynamic>;
+      final String dataMessage = data['message'] as String;
+      final TextDirection textDirection =
+          TextDirection.values[data['textDirection'] as int];
+      final int assertivenessLevel = (data['assertiveness'] as int?) ?? 0;
+      final Assertiveness assertiveness =
+          Assertiveness.values[assertivenessLevel];
+      final CapturedAccessibilityAnnouncement announcement;
+
+      announcement = CapturedAccessibilityAnnouncement(
+          dataMessage, textDirection,
+          assertiveness: assertiveness);
+      _announcements ??= <CapturedAccessibilityAnnouncement>[];
+      _announcements!.add(announcement);
+    }
+  }
+
   Future<void> _runTest(
     Future<void> Function() testBody,
     VoidCallback invariantTester,
     String description,
   ) {
     assert(description != null);
     assert(inTest);
+
+    if (TestDefaultBinaryMessengerBinding.instance!.defaultBinaryMessenger
```

Comment:
Consider leaving a comment explaining what this does.

Reply:
Done!